python/test/unit/language/test_core.py

```diff
@@ -2004,6 +2004,13 @@ def roll(a1, b1_last, b1_cur, a2, b2_last, b2_cur):
 @pytest.mark.parametrize("op, dtype_str, shape, axis, reverse, num_warps", scan_configs + negative_config)
 def test_scan2d(op, dtype_str, shape, axis, reverse, num_warps, device):
     check_type_supported(dtype_str, device)
+    numpy_dtype_str = 'float32' if dtype_str == 'bfloat16' else dtype_str
+
+    if dtype_str == 'bfloat16':
+        if (op == "sum" or op == 'linear_recurrence' or op == 'cumsum'):
+            pytest.skip("FIXME: Skipping " + op + " scan on bfloat16 due to accuracy bugs")
+        if op == 'cummax' and shape[0] * shape[1] >= 1024:
+            pytest.skip("FIXME: Skipping " + op + " scan on bfloat16 due to accuracy bugs under large shape")
```

Comment:
I think those are some other bugs lurking......
Should I create another issue to track them?

Reply:
After some investigation, it turns out that I should pass 'bfloat16' rather than float32 to `numpy_random` to get inputs flaot32 properly truncated(the fp32 will be changed to torch bf16 by `to_triton`), and then the tests work.